packages/mds-api-server/middleware/http-context.ts

```diff
@@ -0,0 +1,10 @@
+import httpContext from 'express-http-context'
+import express from 'express'
+import { ApiRequest, ApiResponse } from '../@types'
+
+export const HttpContextMiddleware = () => httpContext.middleware
+
+export const RequestIdMiddleware = () => (req: ApiRequest, res: ApiResponse, next: express.NextFunction) => {
+  httpContext.set('x-request-id', res.get('x-request-id'))
```

Comment:
Should we assign an id if the gateway didn't?
```suggestion
  httpContext.set('x-request-id', res.get('x-request-id') ?? uuid())
```

Reply:
Yeah, I suppose we could, but I worry that there may be a time when the gateway _doesn't_ add it, and we don't realize that until we're tracing down a request and expecting to see that ID appear in the proxy logs... Issue would be more obvious if we don't have the ID included at all unless it's set by the gateway IMO

Reply:
Given that, maybe set a static warning string?  "WARNING - No X-Request-ID found" something like that?

Reply:
Yeah, I can add something like that.